yktrace/src/hwt/mapper.rs

```diff
@@ -129,15 +129,24 @@ impl HWTMapper {
                 }
             } else {
                 for irblock in irblocks.into_iter() {
-                    // A basic block may map to >1 *machine* basic block. If we see the same block
-                    // repeated, then we need to check if we re-executing the same basic block or
-                    // if we are executing next machine basic block for that same block. In the
-                    // latter case, we mustn't record the block in the trace again.
-                    if !(ret_irblocks.last() == Some(&irblock)
-                        && prev_block.as_ref().map(|x| x.first_instr())
-                            != Some(block.first_instr()))
-                    {
-                        ret_irblocks.push(irblock);
+                    if let Some(irblock) = irblock {
+                        // A basic block may map to >1 *machine* basic block. If we see the same
+                        // block repeated, then we need to check if we re-executing the same basic
```

Comment:
I think you need to specify which block type is which explicitly: I'm not sure what "the same basic block" refers to.